testdata/p4tc_samples_outputs/add_entry_1_example_parser.h

```diff
@@ -65,3 +65,36 @@ REGISTER_TABLE(hdr_md_cpumap, BPF_MAP_TYPE_PERCPU_ARRAY, u32, struct hdr_md, 2)
 BPF_ANNOTATE_KV_PAIR(hdr_md_cpumap, u32, struct hdr_md)
 REGISTER_END()
 
+u32 getPrimitive32(u8 *a) {
+   return  ((((u32)a[2]) <<16) | (((u32)a[1]) << 8) | a[0]);
+}
+u64 getPrimitive64(u8 *a, int size) {
+   if(size < 40) {
+       return  ((((u64)a[4]) << 32) | (((u64)a[3]) << 24) | (((u64)a[2]) << 16) | (((u64)a[1]) << 8) | a[0]);
+   } else {
+       if(size < 48) {
+           return  ((((u64)a[5]) << 40) | (((u64)a[4]) << 32) | (((u64)a[3]) << 24) | (((u64)a[2]) << 16) | (((u64)a[1]) << 8) | a[0]);
+       } else {
+           return  ((((u64)a[6]) << 48) | (((u64)a[5]) << 40) | (((u64)a[4]) << 32) | (((u64)a[3]) << 24) | (((u64)a[2]) << 16) | (((u64)a[1]) << 8) | a[0]);
+       }
+   }
+}
+void storePrimitive32(u8 *a, u32 value) {
+   a[0] = (u8)(value);
+   a[1] = (u8)(value >> 8);
+   a[2] = (u8)(value >> 16);
+}
+void storePrimitive64(u8 *a, int size, u64 value) {
+   a[0] = (u8)(value);
+   a[1] = (u8)(value >> 8);
+   a[2] = (u8)(value >> 16);
+   a[3] = (u8)(value >> 24);
+   a[4] = (u8)(value >> 32);
+   if (size < 48) {
+       a[5] = (u8)(value >> 40);
+   }
+   if (size < 56){
+       a[6] = (u8)(value >> 48);
+   }
```

Comment:
In case you have  48-bit value, this will overwrite the 7th byte which might correspond to another header
Also if the size is 48, you should overwrite the 6th byte

```suggestion
   if (size <= 48) {
       a[5] = (u8)(value >> 40);
   }
   if (size > 48 && size =< 56){
       a[6] = (u8)(value >> 48);
   }
```

You don't need to necessarily implement it like this, it's just to illustrate the idea